packages/template-retail-react-app/jest-setup.js

```diff
@@ -9,6 +9,63 @@ const mockConfig = require(path.join(__dirname, 'config/mocks/default.js'))
 require('raf/polyfill') // fix requestAnimationFrame issue with polyfill
 require('@testing-library/jest-dom/extend-expect')
 const {Crypto} = require('@peculiar/webcrypto')
+const {setupServer} = require('msw/node')
+const {rest} = require('msw')
+const {
+    mockCategory,
+    mockedRegisteredCustomer,
+    exampleTokenReponse
+} = require('./app/commerce-api/mock-data')
+
+/**
+ * Set up an API mocking server for testing purposes.
+ * This mock server includes the basic oauth flow endpoints.
+ */
+export const setupMockServer = () => {
+    return setupServer(
+        rest.post('*/oauth2/authorize', (req, res, ctx) => res(ctx.delay(0), ctx.status(200))),
+        rest.get('*/oauth2/authorize', (req, res, ctx) => res(ctx.delay(0), ctx.status(200))),
+        rest.post('*/oauth2/login', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(mockedRegisteredCustomer))
+        ),
+        rest.get('*/oauth2/logout', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(exampleTokenReponse))
+        ),
+        rest.get('*/customers/:customerId', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(mockedRegisteredCustomer))
+        ),
+        rest.post('*/sessions', (req, res, ctx) => res(ctx.delay(0), ctx.status(200))),
+        rest.post('*/oauth2/token', (req, res, ctx) =>
+            res(
+                ctx.delay(0),
+                ctx.json({
+                    customer_id: 'test',
+                    access_token: 'testtoken',
```

Comment:
instead of mocking `isTokenValid`, we could generate a valid jwt and paste it in here.